packages/sidecar/src/main.ts

```diff
@@ -50,360 +52,175 @@ type SideCarOptions = {
-      const headers = {
-        ...CORS_HEADERS,
-        ...SPOTLIGHT_HEADERS,
-      };
-      for (const [header, value] of Object.entries(headers)) {
-        res.setHeader(header, value);
-      }
-      if (req.method === "OPTIONS") {
-        res.writeHead(204, {
-          "Cache-Control": "no-cache",
-        });
-        res.end();
-        return;
+const contextId = "some-unique-id";
+const transport = new StreamableHTTPTransport();
+const mcp = createMcpInstance();
+
+export const app = new Hono<HonoEnv>()
+  .use(contextStorage(), async (c, next) => {
+    c.header("X-Powered-By", SERVER_IDENTIFIER);
+
+    c.set("contextId", contextId);
+    c.set("transport", transport);
+    await next();
+  })
+  .all(
+    "/mcp",
+    async (c, next) => {
+      if (!mcp.isConnected()) {
+        // Connecting the MCP with the transport
+        await mcp.connect(c.get("transport"));
       }
-      return handler(req, res, pathname, searchParams);
-    },
-    { name: "enableCORS", op: "sidecar.http.middleware.cors" },
-  );
 
-const streamRequestHandler = (buffer: MessageBuffer<Payload>, incomingPayload?: IncomingPayloadCallback) => {
-  return function handleStreamRequest(
-    req: IncomingMessage,
-    res: ServerResponse,
-    pathname?: string,
-    searchParams?: URLSearchParams,
-  ): void {
-    if (
-      req.method === "GET" &&
-      req.headers.accept &&
-      req.headers.accept === "text/event-stream" &&
-      pathname === "/stream"
-    ) {
-      res.writeHead(200, {
-        "Content-Type": "text/event-stream",
-        "Cache-Control": "no-cache",
-        Connection: "keep-alive",
-      });
-      res.flushHeaders();
-      // Send something in the body to trigger the `open` event
-      // This is mostly for Firefox -- see getsentry/spotlight#376
-      res.write("\n");
-
-      const useBase64 = searchParams?.get("base64") != null;
-      const base64Indicator = useBase64 ? ";base64" : "";
-      const dataWriter = useBase64
-        ? (data: Buffer) => res.write(`data:${data.toString("base64")}\n`)
-        : (data: Buffer) => {
-            // The utf-8 encoding here is wrong and is a hack as we are
-            // sending binary data as utf-8 over SSE which enforces utf-8
-            // encoding. This is only for backwards compatibility
-            for (const line of data.toString("utf-8").split("\n")) {
-              // This is very important - SSE events are delimited by two newlines
-              res.write(`data:${line}\n`);
-            }
-          };
+      await next();
+    },
+    async c => {
+      const response = await c.get("transport").handleRequest(c, await c.req.json());
+      return response;
+    },
+  )
+  .get("/health", c => c.text("OK"))
+  .use(cors(), async (_c, next) => {
+    await startSpan({ name: "enableCORS", op: "sidecar.http.middleware.cors" }, async () => await next());
+  })
```

Comment:
Not sure what we are doing here? Looks like we randomly create a span named `enableCORS` but not anywhere near the actual CORS enabling code?


Aside, just wondering if this would fly if this code was actually useful:

```suggestion
  .use(cors(), (_c, next) => {
    startSpan({ name: "enableCORS", op: "sidecar.http.middleware.cors" }, () => next());
  })
```